### DISCUSSION TITLE
error 10013

### DISCUSSION THREAD

**Discussion by bogdankol** [denoland/deno#16181] [Q&A] [answered]
Hi!
I'm trying to run this code in Deno but the result is: Uncaught (in promise) PermissionDenied: An attempt was made to access a socket in a way forbidden by its access permissions. (os error 10013)

```ts
  import { serve } from "https://deno.land/std@0.150.0/http/server.ts";
  import { Server } from "https://deno.land/x/socket_io@0.1.1/mod.ts";

  const io = new Server();

  io.on("connection", (socket) => {
    console.log(`socket ${socket.id} connected`);

    socket.emit("hello", "world");

    socket.on("disconnect", (reason) => {
      console.log(`socket ${socket.id} disconnected due to ${reason}`);
    });
  });

  await serve(io.handler(), {
    port: 3000,
  });
```

And before that code I tried to run 
```ts
   import { serve } from "https://deno.land/std@0.50.0/http/server.ts";
  
   const server = serve({ port: 8008 });
   console.log("http://localhost:8008/");
  
   for await (const req of server) {
   
    if (req.url === '/') {
    req.respond({
      status: 200,
      body: await Deno.open('./public/index.html')
   });
  }
    
  }
```

and it shows the same error. Is there any way to fix this or to run deno properly?

**Comment by kitsonk** [comment, marked as answer]
This is not an issue with Deno, or how you are running `deno`.  The error originates from the underlying operating system indicating that environment/users that `deno` is running under does not have permissions to open a socket to listen for HTTP connections.

**Reply by bogdankol** [reply]
Thank you! Could you please tell me a way to allow socket listening (win10)?

**Reply by Re0niDo** [reply]
Ran into the same issue (also Windows 10) and found this discussion. Definetely not an issue with Deno, but I'll leave a solution that helped me here in case someone also finds this.

The solution was to change the port. To fix the two code examples in the first post you should change `port: 3000` and `port: 8008` to, for example, `port: 10001`.

In my case I was looking at Fresh. I created a new app (`deno run -Ar jsr:@fresh/init`), then when I ran `deno tash dev` I encountered the same error. In my case adding `server: { port: 10001 }` to `vite.config.ts` fixed the problem.
